src/order/conditionally_complete_lattice.lean

```diff
@@ -642,51 +664,90 @@ open_locale classical
 
 variables [conditionally_complete_linear_order_bot α]
 
-lemma has_lub (s : set (with_top α)) : ∃a, is_lub s a :=
+/-- The Sup of a non-empty set is its least upper bound for a conditionally
+complete lattice with a top. -/
+lemma is_lub_Sup' {β : Type*} [conditionally_complete_lattice β]
+  {s : set (with_top β)} (hs : s ≠ ∅) : is_lub s (Sup s) :=
 begin
-  by_cases hs : s = ∅, { subst hs, exact ⟨⊥, is_lub_empty⟩, },
-  rcases ne_empty_iff_exists_mem.1 hs with ⟨x, hxs⟩,
-  by_cases bnd : ∃b:α, ↑b ∈ upper_bounds s,
-  { rcases bnd with ⟨b, hb⟩,
-    have bdd : bdd_above {a : α | ↑a ∈ s}, from ⟨b, assume y hy, coe_le_coe.1 $ hb hy⟩,
-    refine ⟨(Sup {a : α | ↑a ∈ s} : α), _, _⟩,
-    { assume a has,
-      rcases (le_coe_iff _ _).1 (hb has) with ⟨a, rfl, h⟩,
-      exact (coe_le_coe.2 $ le_cSup bdd has) },
-    { assume a hs,
-      rcases (le_coe_iff _ _).1 (hb hxs) with ⟨x, rfl, h⟩,
-      refine (coe_le_iff _ _).2 (assume c hc, _), subst hc,
-      exact (cSup_le (ne_empty_of_mem hxs) $ assume b (hbs : ↑b ∈ s), coe_le_coe.1 $ hs hbs), } },
-  exact ⟨⊤, assume a _, le_top, assume a,
-    match a with
-    | some a, ha := (bnd ⟨a, ha⟩).elim
-    | none,   ha := _root_.le_refl ⊤
-    end⟩
+  split,
+  { show ite _ _ _ ∈ _,
+    split_ifs,
+    { intros _ _, exact le_top},
+    { rintro (⟨⟩|a) ha, contradiction,
+      apply some_le_some.2,
+      exact le_cSup h_1 ha},
+    { intros _ _, exact le_top}},
+  { show ite _ _ _ ∈ _,
+    split_ifs,
+    { rintro (⟨⟩|a) ha, exact _root_.le_refl _,
+      exfalso, apply not_top_le_coe a, exact ha h},
+    { rintro (⟨⟩|b) hb, exact le_top,
+      apply some_le_some.2,
+      refine cSup_le _ _,
+      { refine mt _ h,
+        rw ne_empty_iff_exists_mem at hs,
+        intro h2,
+        rcases hs with ⟨⟨⟩|b, hb⟩, assumption,
+        exfalso, revert h2, exact ne_empty_of_mem hb},
+      { intros a ha, exact some_le_some.1 (hb ha)}},
+    { rintro (⟨⟩|b) hb, exact _root_.le_refl _,
+      exfalso, apply h_1, use b, intros a ha, exact some_le_some.1 (hb ha)}}
 end
 
-lemma has_glb (s : set (with_top α)) : ∃a, is_glb s a :=
+lemma is_lub_Sup (s : set (with_top α)) : is_lub s (Sup s) :=
 begin
-  by_cases hs : ∃x:α, ↑x ∈ s,
-  { rcases hs with ⟨x, hxs⟩,
-    refine ⟨(Inf {a : α | ↑a ∈ s} : α), _, _⟩,
-    exact (assume a has, (coe_le_iff _ _).2 $ assume x hx, cInf_le (bdd_below_bot _) $
-      show ↑x ∈ s, from hx ▸ has),
-    { assume a has,
-      rcases (le_coe_iff _ _).1 (has hxs) with ⟨x, rfl, h⟩,
-      exact (coe_le_coe.2 $ le_cInf (ne_empty_of_mem hxs) $
-        assume b hbs, coe_le_coe.1 $ has hbs) } },
-  exact ⟨⊤, assume a, match a with
-    | some a, ha := (hs ⟨a, ha⟩).elim
-    | none,   ha := _root_.le_refl _
-    end,
-    assume a _, le_top⟩
+  by_cases hs : s = ∅,
+  { rw hs,
+    show is_lub ∅ (ite _ _ _),
+    split_ifs,
+    { cases h},
+    { rw [preimage_empty, cSup_empty], exact is_lub_empty},
+    { exfalso, apply h_1, use ⊥, rintro a ⟨⟩}
+  },
```

Comment:
move to previous line, and add spaces before the braces on the previous lines.

Reply:
done